crates/utils/src/dir.rs

```diff
@@ -0,0 +1,23 @@
+use std::{fs, io, path::PathBuf};
+
+use directories::BaseDirs;
+
+/// Creates a `.ream` directory in the system's data directory if it doesn't exist.
+///
+/// # Returns
+///
+/// Returns the path to the `.ream` directory on success, or an `io::Error` if it fails.
+pub fn create_ream_dir() -> io::Result<PathBuf> {
+    if let Some(base_dirs) = BaseDirs::new() {
+        let ream_dir = base_dirs.data_dir().join(".ream");
```

Comment:
```suggestion
/// Creates a `ream` directory in the system's data directory if it doesn't exist.
///
/// # Returns
///
/// Returns the path to the `.ream` directory on success, or an `io::Error` if it fails.
pub fn create_ream_dir() -> io::Result<PathBuf> {
    if let Some(base_dirs) = BaseDirs::new() {
        let ream_dir = base_dirs.data_dir().join("ream");
```

Why are we trying to make the ream directory hidden? It should be a public folder